tracer/src/Datadog.Trace.Annotations/TraceAttribute.cs

```diff
@@ -0,0 +1,29 @@
+// <copyright file="TraceAttribute.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+
+namespace Datadog.Trace.Annotations
+{
+    /// <summary>
+    /// Attribute that marks the decorated method to be instrumented
+    /// by Datadog automatic instrumentation.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = false)]
```

Comment:
Can we use this on properties too, or only methods right now?

Reply:
Right now it's only methods, though I'm not sure how well the attribute would map to properties because a property can map to both a getter method and a setter method, and I'm not sure we'd want to set the same `ResourceName` for both. I just realized that I can decorate a property accessor method like the following (and added this to my integration tests in 46ae5f8dd067fd13db2b590f9a283f7ed6c1edea), so I think this will handle the case where users want to decorate property usages. Does that sound good?

```csharp
class TestType
{
    public string Name
    {
        [Trace]
        get;
        [Trace]
        set;
    }
}
```

Reply:
Not needed for this PR, but if we allow this:
```csharp
class TestType
{
    [Trace]
    public string Name { get; set; }
}
```

Could the resource names be `get_Name` and `set_Name`, since those are the actual method names behind-the-scenes? Hmm, but I guess that gets weird is they override the resource name in the attribute.

Reply:
Yeah that's what I was trying to avoid, if the user decided to change the resource name. That's why I'm proposing we limit it to just methods instead of trying to make it work for properties. Plus property support on the native side would be more complicated 😅